exercises/00-init/src/modules/root/components/users-list.js

```diff
@@ -3,12 +3,12 @@ import PropTypes from 'prop-types';
 
 const UsersList = (props) => (
     <div>
-    <button onClick={ () => props.addUser('Arya', 'Stark')}>Add Arya</button>
-    <button onClick={ () => props.addUser('Daenerys', 'Targaryen')}>Add Daenerys</button>
+    <button onClick={ () => props.addUser( {firstName:'Arya', lastName:'Stark'} )}>Add Arya</button>
+    <button onClick={ () => props.addUser( {firstName:'Daenerys', lastName:'Targaryen'} )}>Add Daenerys</button>
     <br/>
-    {props.users.length === 0
+    {props.users.users.length === 0
         ? "No users" 
-        : props.users.map( ({id, firstName, lastName}) => 
+        : props.users.users.map( ({id, firstName, lastName}) => 

```

Comment:
Changes to this file should not really be necessary:
1. instead of `props.users.users` you can directly pass `store.getState().users.users` to the `<Root>` component in `index.js` (granted, it is a bit confusing, in practice I would probably call the list something like `usersList` so that it would then be `store.getState().users.usersList`). Actually, if you check the console, you will see that the propTypes checks are failing for both `Root` and `UsersList` - you specified that you expect `users` to be an array of objects, but you are actually passing the entire `users` substore, which is an object.
2. You don't need to modify the signature of the `addUser` method. You could simply define the `dispatchAddUser` in `index.js` as follows:
```
const dispatchAddUser = (firstName, lastName) => {
  store.dispatch(UsersActions.Creators.addUser({ firstName, lastName })
}
```

Reply:
Yeah, I didn't like users.users. On my first try I removed 'title' from the users reducer, but then I didn't have anywhere to put the title, and I didn't want to create another reducer, or move it outside the state... Renaming the user list to userList makes it better.
Fixed it all.